ntoskrnl/se/debug.c

```diff
@@ -0,0 +1,479 @@
+
+    DPRINT1("================== SECURITY DESCRIPTOR DUMP INFO ==================\n");
+    DPRINT1("SD pointer -> %p\n", SecurityDescriptor);
+    DPRINT1("SD Revision -> %u\n", SecurityDescriptor->Revision);
+    DPRINT1("SD Control:\n");
+    SepDumpSdControlInfo(SecurityDescriptor->Control);
+
+    /* Dump the Owner SID if the SD belongs to an owner */
+    if (OwnerSid)
+    {
+        RtlConvertSidToUnicodeString(&SidString, OwnerSid, TRUE);
+        DPRINT1("SD Owner SID -> %wZ\n", &SidString);
+        RtlFreeUnicodeString(&SidString);
+    }
+
+    /* Dump the Group SID if the SD belongs to a group */
+    if (GroupSid)
+    {
+        RtlConvertSidToUnicodeString(&SidString, GroupSid, TRUE);
+        DPRINT1("SD Group SID -> %wZ\n", &SidString);
+        RtlFreeUnicodeString(&SidString);
+    }
+
+    /* Dump the ACL contents of SACL if this SD has one */
+    if (Sacl)
+    {
+        SepDumpAclInfo(Sacl, TRUE);
+    }
+
+    /* Dump the ACL contents of DACL if this SD has one */
+    if (Dacl)
+    {
+        SepDumpAclInfo(Dacl, FALSE);
+    }
+}
+
+/**
+ * @brief
+ * Dumps debug information of an access
+ * token to the debugger.
+ *
+ * @param[in] Token
+ * Pointer to an access token of which
+ * debug data is to be dumped from.
+ *
+ * @return
+ * Nothing.
+ */
+VOID
+SepDumpTokenDebugInfo(
+    _In_ PTOKEN Token)
+{
+    UNICODE_STRING SidString;
+
+    /* Dump relevant token info */
+    DPRINT1("================== ACCESS TOKEN DUMP INFO ==================\n");
+    DPRINT1("Token pointer -> %p\n", Token);
+    DPRINT1("Token source -> %s\n", Token->TokenSource.SourceName);
+    DPRINT1("Token source identifier (LowPart) -> %lu\n", Token->TokenSource.SourceIdentifier.LowPart);
+    DPRINT1("Token source identifier (HighPart) -> %lu\n", Token->TokenSource.SourceIdentifier.HighPart);
+
+    RtlConvertSidToUnicodeString(&SidString, Token->PrimaryGroup, TRUE);
+    DPRINT1("Token primary group SID -> %wZ\n", &SidString);
+    RtlFreeUnicodeString(&SidString);
+
+    DPRINT1("Token user and groups SIDs:\n");
+    SepDumpSidsOfToken(Token->UserAndGroups, Token->UserAndGroupCount);
+
+    if (SeTokenIsRestricted(Token))
+    {
+        DPRINT1("Token restricted SIDs:\n");
+        SepDumpSidsOfToken(Token->RestrictedSids, Token->RestrictedSidCount);
+    }
+}
+
+/**
+ * @brief
+ * Dumps security access rights to the
+ * debugger.
+ *
+ * @param[in] AccessRights
+ * Pointer to a list of access rights
+ * that have been granted, denied and
+ * remaining rights to be dumped.
+ *
+ * @return
+ * Nothing.
+ */
+VOID
+SepDumpAccessRightsStats(
+    _In_ PACCESS_CHECK_RIGHTS AccessRights)
+{
+    DPRINT1("================== ACCESS CHECK RIGHTS STATISTICS ==================\n");
+    DPRINT1("Remaining access rights -> 0x%08lx\n", AccessRights->RemainingAccessRights);
+    DPRINT1("Granted access rights -> 0x%08lx\n", AccessRights->GrantedAccessRights);
+    DPRINT1("Denied access rights -> 0x%08lx\n", AccessRights->DeniedAccessRights);
```

Comment:
(Same comment for all those debug functions). It would be better to use the `DbgPrint` function directly, instead of `DPRINT(1)`, in order to get rid of all those useless file/line number info showing up, that clog the debug output.